lib/maths/common/CQuantileSketch.cc

```diff
@@ -24,8 +24,89 @@
 #include <algorithm>
 #include <cmath>
 #include <limits>
+#include <numeric>
+#include <optional>
 #include <random>
 
+#if defined(__SSE4_2__)
```

Comment:
`xmmintrin.h` is for the original SSE instructions, not SSE 4.2:

```
// instrinsics prototypes
#include <xmmintrin.h>     //SSE
#include <emmintrin.h>     //SSE2
#include <pmmintrin.h>     //SSE3
#include <tmmintrin.h>     //SSSE3
#include <smmintrin.h>     //SSE4.1
#include <nmmintrin.h>     //SSE4.2
```

So it should be this:

```suggestion
#if defined(__SSE__)
```

If that then shows a problem on Windows then I'd rather explicitly exclude Windows rather than use a check for a macro that Windows doesn't have to sneakily exclude it.

Reply:
Windows was a mistake in the test. In this case, I do only use SSE instructions since it compiles after all. I think I'll probably just keep including the minimum set I need, but agree swapping to __SSE__ probably makes sense. (I went with __SSE4_2__ because it is also understood by all our compilers and we compile with SSE4.2 enabled.)

Reply:
I don't think Windows was compiling the code in this branch before.  Windows doesn't have an option to use SSE4.2 instructions unless you are prepared to use AVX.  There's no option in between SSE2 and AVX - see https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=msvc-160.  So I think on Windows checking `__SSE4_2__` without using the `/arch:AVX` compiler option would mean this optimisation wasn't used when it could be.  And we don't want to use `/arch:AVX` because that stops us working on a whole load of AMD processors that some users may still be using.

Reply:
Right that's worth knowing. I updated the ifdef to your suggestion [here](https://github.com/elastic/ml-cpp/pull/2390/commits/f5cec9737cdf6326d6ece2bf39dde4e99ce25468). Note that the real fix for Windows was [this](https://github.com/elastic/ml-cpp/pull/2390/commits/c98cf0c1e5c21ee43210549a03b9f2ad179f234b) one.